ScriptureRenderingPipeline/Webhook.cs

```diff
@@ -23,301 +23,326 @@
-            MemoryStream templateStream = new MemoryStream();
-            await blobClient.DownloadToAsync(templateStream);
-            templateStream.Seek(0, SeekOrigin.Begin);
-            return await new StreamReader(templateStream).ReadToEndAsync();
-        }
-
-    }
+	public static class Webhook
+	{
+		[FunctionName("Webhook")]
+		public static async Task<IActionResult> RunAsync(
+				[HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "webhook")] HttpRequest req,
+				ILogger log)
+		{
+			var connectionString = Environment.GetEnvironmentVariable("ScripturePipelineStorageConnectionString");
+			var outputContainer = Environment.GetEnvironmentVariable("ScripturePipelineStorageOutputContainer");
+			var templateContainer = Environment.GetEnvironmentVariable("ScripturePipelineStorageTemplateContainer");
+			var baseUrl = Environment.GetEnvironmentVariable("ScriptureRenderingPipelineBaseUrl");
+			var userToRouteResourcesTo = Environment.GetEnvironmentVariable("ScriptureRenderingPipelineResourcesUser");
+			string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+			WebhookEvent webhookEvent = JsonConvert.DeserializeObject<WebhookEvent>(requestBody);
+
+			DateTime timeStarted = DateTime.Now;
+
+
+			// validate
+
+			if (webhookEvent == null)
+			{
+				return new BadRequestObjectResult("Invalid webhook request");
+			}
+
+			if (req.Headers.ContainsKey("X-GitHub-Event"))
+			{
+				if (req.Headers["X-GitHub-Event"] != "push")
+				{
+					return new OkObjectResult("Not converting because this isn't a push event");
+				}
+			}
+
+			if (webhookEvent.commits == null)
+			{
+				return new BadRequestObjectResult("Missing commits from webhook event");
+			}
+
+			log.LogInformation("Starting webhook for {repoName}", webhookEvent.repository.FullName);
+
+			// download repo
+
+			log.LogInformation($"Downloading repo");
+
+			using var httpClient = new HttpClient();
+			var downloadProjectPageTemplateTask = GetTemplateAsync(connectionString, templateContainer, "project-page.html");
+			var downloadPrintPageTemplateTask = GetTemplateAsync(connectionString, templateContainer, "print.html");
+			var result = await httpClient.GetAsync($"{webhookEvent.repository.HtmlUrl}/archive/master.zip");
+			if (result.StatusCode == HttpStatusCode.NotFound)
+			{
+				return new BadRequestObjectResult("Can't download source zip, probably an empty repo");
+			}
+			var zipStream = await result.Content.ReadAsStreamAsync();
+			var fileSystem = new ZipFileSystem(zipStream);
+
+			var repoType = RepoType.Unknown;
+			var isBTTWriterProject = false;
+			var outputDir = Utils.CreateTempFolder();
+			string exceptionMessage = null;
+			var title = "";
+			string template = null;
+			var converterUsed = "";
+			var languageName = string.Empty;
+			var resourceName = string.Empty;
+			var languageDirection = "ltr";
+			var languageCode = string.Empty;
+			JsonElement appsMeta = new JsonElement();

```

Comment:
I recommend sticking with precedent here and using "var" instead of "JsonElement."  The compiler should be able to infer the type from the "new JsonElement".